azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/domain/NetworkInterfaceCardProperties.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.azurecompute.arm.domain;
+
+import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableList;
+import org.jclouds.javax.annotation.Nullable;
+import org.jclouds.json.SerializedNames;
+
+import java.util.List;
+
+@AutoValue
+public abstract class NetworkInterfaceCardProperties {
+
+   @Nullable
+   public abstract String provisioningState();
+
+   @Nullable
+   public abstract String resourceGuid();
+
+   @Nullable
+   public abstract Boolean enableIPForwarding();
+
+   @Nullable
+   public abstract ImmutableList<IpConfiguration> ipConfigurations();
+
+   @SerializedNames({"provisioningState", "resourceGuid", "enableIPForwarding", "ipConfigurations"})
+   public static NetworkInterfaceCardProperties create(final String provisioningState, final String resourceGuid, final Boolean enableIPForwarding, final List<IpConfiguration> ipConfigurations) {
+      NetworkInterfaceCardProperties.Builder builder = NetworkInterfaceCardProperties.builder()
+              .provisioningState(provisioningState)
+              .resourceGuid(resourceGuid)
+              .enableIPForwarding(enableIPForwarding);
+
+      if (ipConfigurations != null)
+         builder.ipConfigurations(ipConfigurations);
+
+      return builder.build();
+   }
+
+   public static Builder builder() {
+      return new AutoValue_NetworkInterfaceCardProperties.Builder();
+   }
+
+   @AutoValue.Builder
+   public abstract static class Builder {
+      public abstract Builder provisioningState(String provisioningState);
+
+      public abstract Builder resourceGuid(String resourceGuid);
+
+      public abstract Builder enableIPForwarding(Boolean enableIPForwarding);
+
+      public abstract Builder ipConfigurations(List<IpConfiguration> ipConfigurations);
```

Comment:
For builders using lists, we want to enforce immutable lists while keeping clean signatures with `List` interface types. You can use [this pattern](https://github.com/jclouds/jclouds-labs-openstack/blob/master/openstack-heat/src/main/java/org/jclouds/openstack/heat/v1/options/UpdateStack.java#L67-L76) that uses hidden methods in the builder to enforce immutability. Also apply this pattern to all builder methods that receive collection parameters.